mappings/net/minecraft/village/TradeOffer.mapping

```diff
@@ -50,27 +50,77 @@ CLASS net/minecraft/class_1914 net/minecraft/village/TradeOffer
+	METHOD method_18019 copySellItem ()Lnet/minecraft/class_1799;
+		COMMENT Returns a copy of the sell item of this trade offer.
 	METHOD method_19272 getAdjustedFirstBuyItem ()Lnet/minecraft/class_1799;
+		COMMENT Returns a copy of the first buy item of this trade offer,
+		COMMENT with its price adjusted depending on the demand bonus, the
+		COMMENT special price and the price multiplier.
 	METHOD method_19273 setSpecialPrice (I)V
+		COMMENT Sets the special price of this trade offer to {@code specialPrice}.
 		ARG 1 specialPrice
-	METHOD method_19274 updatePriceOnDemand ()V
+	METHOD method_19274 updateDemandBonus ()V
+		COMMENT Updates the demand bonus of this trade offer depending on its
+		COMMENT previous demand bonus, the number of times it has been used and
+		COMMENT its remaining uses.
 	METHOD method_19275 resetUses ()V
+		COMMENT Resets the number of times this trade offer has been used.
 	METHOD method_19276 clearSpecialPrice ()V
+		COMMENT Resets the special price of this trade offer.
 	METHOD method_19277 getSpecialPrice ()I
+		COMMENT Returns the special price of this trade offer. It is used to
+		COMMENT adjust the price of its first buy item.
+		COMMENT
+		COMMENT <p>The less the special price is, the more the price will be low.
 	METHOD method_19278 getPriceMultiplier ()F
+		COMMENT Returns the price multiplier of this trade offer. It is used to
+		COMMENT adjust the price of its first buy item.
 	METHOD method_19279 getMerchantExperience ()I
+		COMMENT Returns the amount of experience that will be given to a merchant
+		COMMENT after this trade offer has been used.
 	METHOD method_21725 getDemandBonus ()I
-	METHOD method_21834 hasAvailableUses ()Z
+		COMMENT Returns the demand bonus of this trade offer. It is used to
+		COMMENT adjust the price of its first buy item.
+		COMMENT
+		COMMENT <p>The more the demand bonus is, the more the price will be high.
+		COMMENT
+		COMMENT <p>It is updated when a villager restocks.
+		COMMENT
+		COMMENT @see #updateDemandBonus()
+	METHOD method_21834 hasBeenUsed ()Z
+		COMMENT Returns whether this trade offer has already been used.
+		COMMENT
+		COMMENT <p>Checks if the number of times this trade offer has been used
+		COMMENT is greater than 0.
 	METHOD method_8244 use ()V
+		COMMENT Increments the number of times this trade offer has been used.
 	METHOD method_8245 increaseSpecialPrice (I)V
+		COMMENT Increases the special price of this trade offer by {@code increment}.
+		COMMENT
+		COMMENT <p>A negative {@code increment} value will decrease the special price.
 		ARG 1 increment
 	METHOD method_8246 getOriginalFirstBuyItem ()Lnet/minecraft/class_1799;
+		COMMENT Returns the first buy item of this trade offer.
 	METHOD method_8247 getSecondBuyItem ()Lnet/minecraft/class_1799;
+		COMMENT Returns the second buy item of this trade offer.
+		COMMENT
+		COMMENT <p>If there is no second buy item, it equals to the {@linkplain net.minecraft.item.ItemStack#EMPTY empty
```

Comment:
```suggestion
		COMMENT <p>If there is no second buy item, this returns the {@linkplain ItemStack#EMPTY empty
```

Reply:
Hm I thought writing the full package name was necessary in link/linkplain?

Reply:
See [here](https://github.com/FabricMC/yarn/blob/0218c2a387a8cca390743a04974db257bcd5e5f8/CONVENTIONS.md#linking-to-other-members) for exceptions